tests/modules/mcuboot/external_flash/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+  mcuboot.external_flash:
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
+    tags: mcuboot external_flash
```

Comment:
to ensure we don't break existing behavior that has been there for a long time, should we have a test case that can be removed when the Kconfig support is removed ?

Reply:
There is already a bunch of `nrf_desktop` samples that does exactly this.

Reply:
I will create a task to ensure that this is done if nrf_desktop migrates before the old way is removed.